### ISSUE TITLE
Workspace Indexing Never Finishes

### ISSUE THREAD

**Issue by jamesward** [aws/aws-toolkit-jetbrains#4721]
I'm on IntelliJ 2024.1.4 on Linux with Amazon Q `3.18-241`.

I've enabled "Workspace index" but after waiting days, I still get this in Q Chat:
> By the way, I'm still indexing this project for full context from your workspace. I may have a better response in a few minutes when it's complete if you'd like to try again then.

I've tried with both the GPU setting on & off.

**Comment by zixlin7** [comment]
Hi, could you pls share logs in your idea.log for us to investigate? thanks

**Comment by jamesward** [comment]
Anything I should filter on as my `idea.log` file is currently pretty large?

**Comment by jamesward** [comment]
Oh looks like the newest versions (`3.20-241`) maybe are making progress.  I opened a projected and asked "@workspace what does this project do?" which returned a response including:
> By the way, I'm still indexing this project for full context from your workspace. I may have a better response in a few minutes when it's complete if you'd like to try again then.
After a few minutes I asked the same questions again and then it worked.  So it looks like this particular issue is solved but one thing to note is that after restarting IntelliJ, and asking the question again I got the "I'm still indexing" answer. So it seems that the index isn't persisted.  Closing this unless you want to have an issue for the lack of persistence to the index.
